### ISSUE TITLE
Unable to build latest master in Fedora 25 with libmypaint from packages

### ISSUE THREAD

**Issue by manuq** [mypaint/mypaint#800]
#### Unable to build latest master in Fedora 25 with libmypaint from packages

Build fails at latest master in Fedora 25 with libmypaint 1.3.0 installed from packages. BUILDING.md says libmypaint should be version 1.3.0 or later. I wonder if that version is still relevant.

#### Basic system details

MyPaint version: git revision 687f3ea9 and libmypaint version 1.3.0 release 1.fc25 installed from the updates DNF repository.
Operating system: Fedora 25
Desktop environment: GNOME

#### Steps to reproduce

```
sudo dnf install libmypaint-devel
python setup.py demo
```
#### Backtraces or error messages

```
[manuq@localhost mypaint]$ LC_ALL=C python setup.py demo
running demo
running build_ext
building 'lib._mypaintlib' extension
swigging lib/mypaintlib.i to lib/mypaintlib_wrap.cpp
swig -python -Wall -noproxydel -c++ -I/usr/lib64/python2.7/site-packages/numpy/core/include -I/usr/include/pygobject-3.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmypaint -I/usr/include/json-c -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DNO_TESTS -o lib/mypaintlib_wrap.cpp lib/mypaintlib.i
gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/lib64/python2.7/site-packages/numpy/core/include -I/usr/include/pygobject-3.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmypaint -I/usr/include/json-c -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/python2.7 -c lib/mypaintlib_wrap.cpp -o build/temp.linux-x86_64-2.7/lib/mypaintlib_wrap.o -Wall -Wno-sign-compare -Wno-write-strings -D_POSIX_C_SOURCE=200809L -DNO_TESTS -g -fopenmp -pthread -O3
In file included from /usr/include/python2.7/pyconfig.h:6:0,
                 from /usr/include/python2.7/Python.h:8,
                 from lib/mypaintlib_wrap.cpp:173:
/usr/include/python2.7/pyconfig-64.h:1199:0: warning: "_POSIX_C_SOURCE" redefined
 #define _POSIX_C_SOURCE 200112L
 
<command-line>:0:0: note: this is the location of the previous definition
In file included from lib/mypaintlib.hpp:18:0,
                 from lib/mypaintlib_wrap.cpp:3140:
lib/tiledsurface.hpp: In member function 'void TiledSurface::set_symmetry_state(bool, float, float, SymmetryType, int)':
lib/tiledsurface.hpp:49:10: error: 'MyPaintSymmetryType' was not declared in this scope
         (MyPaintSymmetryType)symmetry_type, rot_symmetry_lines);
          ^~~~~~~~~~~~~~~~~~~
error: command 'gcc' failed with exit status 1

```

**Comment by odysseywestra** [comment]
There have been recent changes to libmypaint that mypiant-git now requires libmypaint to be build from source.

**Comment by manuq** [comment]
@odysseywestra I will buid libmypaint from source then. Thanks!

**Comment by manuq** [comment]
@odysseywestra I was able to build MyPaint by building libmypaint from source. Libmypaint needs to be configured with  `--prefix=/usr`, otherwise MyPaint raises `ImportError: No module named _mypaintlib` even if I set `PKG_CONFIG_PATH=/usr/local/lib/pkgconfig`.

I see there is some appimage in the source. Have you considered [flatpak](http://flatpak.org/) ? If not I would like to try building a flatpak package of MyPaint.
